components/console_cmd_mqtt/console_mqtt.c

```diff
@@ -0,0 +1,222 @@
+    int nerrors = arg_parse(argc, argv, (void **)&mqtt_args);
+    if (nerrors != 0) {
+        arg_print_errors(stderr, mqtt_args.end, argv[0]);
+        return 1;
+    }
+
+    if (mqtt_args.connect->count > 0) {
+
+        printf("Connect command received:\n");
+        esp_mqtt_client_config_t mqtt_cfg = {
+            .broker.address.uri = CONFIG_MQTT_BROKER_URL
+        };
+
+        printf("broker: %s\n", mqtt_cfg.broker.address.uri);
+
+        client_hndl = esp_mqtt_client_init(&mqtt_cfg);
+        if (client_hndl == NULL) {
+            printf("ERROR: Client init\n");
+            return 1;
+        }
+        /* The last argument may be used to pass data to the event handler, in this example mqtt_event_handler */
+        esp_mqtt_client_register_event(client_hndl, ESP_EVENT_ANY_ID, mqtt_event_handler, NULL);
+        esp_mqtt_client_start(client_hndl);
+
+    } else if (mqtt_args.publish->count > 0) {
+
+        printf("Publish command received:\n");
+        if ((mqtt_args.topic->count <= 0) || (mqtt_args.data->count <= 0)) {
+            printf(PUBLISH_HELP_MSG);
+        }
+        printf("  Topic: %s\n", mqtt_args.topic->sval[0]);
+
+        msg_id = esp_mqtt_client_publish(client_hndl,
+                                         mqtt_args.topic->sval[0],
+                                         mqtt_args.data->sval[0],
+                                         0, 1, 0);
+        ESP_LOGI(TAG, "Publish successful, msg_id=%d, topic=%s, data=%s",
+                 msg_id, mqtt_args.topic->sval[0], mqtt_args.data->sval[0]);
+
+    } else if (mqtt_args.subscribe->count > 0) {
+        if (mqtt_args.topic->count <= 0) {
+            printf(SUBSCRIBE_HELP_MSG);
+        }
+        char *topic = (char *)mqtt_args.topic->sval[0];
+        printf("subscribe command received:\n");
+        printf("  Topic: %s\n", topic);
+        msg_id = esp_mqtt_client_subscribe(client_hndl, topic, 0);
+        ESP_LOGI(TAG, "sent subscribe successful, msg_id=%d", msg_id);
+
+    } else if (mqtt_args.unsubscribe->count > 0) {
+        if (mqtt_args.topic->count <= 0) {
+            printf(UNSUBSCRIBE_HELP_MSG);
+        }
+        printf("unsubscribe command received:\n");
+        printf("  Topic: %s\n", mqtt_args.topic->sval[0]);
+        msg_id = esp_mqtt_client_unsubscribe(client_hndl, mqtt_args.topic->sval[0]);
+        ESP_LOGI(TAG, "sent unsubscribe successful, msg_id=%d", msg_id);
+
+    } else if (mqtt_args.list_subscriptions->count > 0) {
+        printf("TBD: list_subscriptions command\n");
+
+    } else if (mqtt_args.disconnect->count > 0) {
+        printf("Disconnect command received:\n");
+        esp_mqtt_client_disconnect(client_hndl);
+
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Registers the mqtt command.
+ *
+ * @return
+ *          - esp_err_t
+ */
+esp_err_t console_cmd_mqtt_register(void)
+{
+    esp_err_t ret;
+
+    mqtt_args.connect = arg_lit0("C", "connect", "Connect to a broker");
```

Comment:
General question. Can we connect specific broker using console command line? or the broker should be only in sdkconfig?

Reply:
With the current version we can now pass the broker uri with -h.